WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/StatsWidgetConfigureFragment.kt

```diff
@@ -12,22 +14,49 @@ import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.ViewModelProviders
 import dagger.android.support.DaggerFragment
-import kotlinx.android.synthetic.main.stats_views_widget_configure_fragment.*
+import kotlinx.android.synthetic.main.stats_widget_configure_fragment.*
 import org.wordpress.android.R
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureFragment.ViewType.ALL_TIME_VIEWS
+import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureFragment.ViewType.WEEK_VIEWS
+import org.wordpress.android.util.image.ImageManager
 import javax.inject.Inject
 
-class StatsViewsWidgetConfigureFragment : DaggerFragment() {
+class StatsWidgetConfigureFragment : DaggerFragment() {
     @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
     @Inject lateinit var viewsWidgetUpdater: ViewsWidgetUpdater
-    private lateinit var viewModel: StatsViewsWidgetConfigureViewModel
+    @Inject lateinit var appPrefsWrapper: AppPrefsWrapper
+    @Inject lateinit var siteStore: SiteStore
+    @Inject lateinit var imageManager: ImageManager
+    private lateinit var viewModel: StatsWidgetConfigureViewModel
+    private lateinit var viewType: ViewType
+
+    override fun onInflate(context: Context?, attrs: AttributeSet?, savedInstanceState: Bundle?) {
+        super.onInflate(context, attrs, savedInstanceState)
+        activity?.let {
+            val a = it.obtainStyledAttributes(attrs, R.styleable.statsWidget)
+            val views = a.getInt(R.styleable.statsWidget_viewType, -1)
+            viewType = when (views) {
+                -1 -> throw IllegalArgumentException("The view type needs to be specified on the fragment")
```

Comment:
I assume -1 case will be covered by else? If so maybe we can remove it :)

Reply:
you're right 👍